flow/flow.go

```diff
@@ -251,7 +251,10 @@ func NewFlow(r *consent.LoginRequest) *Flow {
 }
 
 func (f *Flow) HandleLoginRequest(h *consent.HandledLoginRequest) error {
-	if f.State != FlowStateLoginInitialized {
+
+	if f.State == FlowStateLoginUnused {
+		return nil
+	} else if f.State != FlowStateLoginInitialized {
```

Comment:
Would it make sense to use the same logic as in the consent flow?

```

	if f.State != FlowStateConsentInitialized && f.State != FlowStateConsentUnused && f.State != FlowStateConsentError {
		return errors.Errorf("invalid flow state: expected %d/%d/%d, got %d", FlowStateConsentInitialized, FlowStateConsentUnused, FlowStateConsentError, f.State)
	}
```

Reply:
On a second look I am still confused though. The 409 error does not happen in this logic here, it comes from the database or explicitly returning `x.ErrConflict`.

I then executed the tests included in this PR without the changes above, and the tests pass, indicating that the change has not the desired effect and the code on this branch already behaves the way we want it to?

Reply:
```suggestion
        if f.State != FlowStateLoginInitialized && f.State != FlowStateLoginUnused {
		return errors.Errorf("invalid flow state: expected %d/%d, got %d", FlowStateLoginInitialized, FlowStateLoginUnused, f.State)
	}
```

I don't understand exactly what you're saying is working or not but the suggestion of changing it to be more similar to the `HandleConsentRequest` function makes sense to me and if I'm not wrong it's logically equivalent.

Reply:
Thank you for getting back :) I ran the tests you provided but changed the code in flow.go back to what it was before your changes, and the tests you wrote still pass, making me wonder whether we need this change in flow.go?

Reply:
Hmm they fail locally for me but I could be doing something wrong. I created https://github.com/ory/hydra/pull/3085 to attempt to run tests in the pipeline but if there is a PR pipeline I didn't manage to trigger it.

Reply:
My bad @jonkjenn ! You're right :)

Reply:
@jonkjenn cc @aeneasr you are right that there are some opportunities for more consistency checks. When I introduced Flow, I initially tried to validate everything. But it turned out that some tests depend on imperfections in the validation, and that updating them would be a lot of work, so I changed my strategy and just made the minimum necessary changes.